src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -179,6 +168,65 @@ public void shutdown() {
         }
     }
 
+    public class CertificateAuthorityChangeHandler {
+
+        private BatchedSubscriber subscriber;
+
+        @SuppressWarnings({"unchecked", "PMD.UnusedFormalParameter"})
+        private void onCAChange(WhatHappened what, Set<Node> whatChanged) {
+            Topic caTopic;
+            try {
+                caTopic = findCATopic();
+            } catch (ServiceLoadException e) {
+                serviceErrored(e);
+                return;
+            }
+
+            List<String> caCerts = (List<String>) caTopic.toPOJO();
+            if (Utils.isEmpty(caCerts)) {
+                return;
+            }
+
+            logger.atDebug().kv("numCaCerts", caCerts.size()).log("CA update received");
+            try {
+                mqttClientKeyStore.updateCA(caCerts);
+            } catch (IOException | CertificateException | KeyStoreException e) {
+                serviceErrored(e);
+            }
+        }
+
+        /**
+         * Begin listening and responding to CDA CA changes.
+         *
+         * <p>This operation is idempotent.
+         *
+         * @throws ServiceLoadException if CDA service could not be loaded
+         */
+        public void start() throws ServiceLoadException {
+            if (subscriber == null) {
+                subscriber = new BatchedSubscriber(findCATopic(), this::onCAChange);
+            }
+            subscriber.subscribe();
+        }
+
+        private Topic findCATopic() throws ServiceLoadException {
```

Comment:
for a nicer API, I'd probably just catch the error in here, and return null, or an Optional<Topic>

Reply:
we'd need a way to propagate the error up though so we can quickly/cleanly exit `startup` though?  that feels a little convoluted with Optional/null.  just pushed up a commit to try it out, I'm not crazy about it, but wanted to get your thoughts if there's a cleaner way

Reply:
We're trying to decouple bridge from CDA ultimately, so I don't think that this should really cause the service to error at all. At this time we know that it can never fail because bridge has a hard dependency on CDA. But I'd like to start the scaffolding for decoupling them by not blowing up if the service doesn't exist.

Reply:
Ahh okay that makes sense, will remove the error and consider maybe a warn log

Reply:
done